keps/sig-apps/4026-crojob-scheduled-timestamp-annotation/README.md

```diff
@@ -0,0 +1,289 @@
+
+##### e2e tests
+
+E2E tests will not provide any additional coverage that isn't already covered by unit + integration tests, since we are simply adding an annotation, so no e2e tests will be necessary for this change.
+
+### Graduation Criteria
+
+The feature will be released directly in Beta state since there is no benefit in having an alpha release, since we are simply adding a new annotation so there is very little risk.
+
+#### Beta
+
+- Feature implemented behind the `CronJobsScheduledAnnotation` feature gate.
+- Unit and integration tests passing.
+
+#### GA
+
+Fix any potentially reported bugs.
+
+### Upgrade / Downgrade Strategy
+
+No changes required to existing cluster to use this feature.
+
+### Version Skew Strategy
+
+N/A. This feature doesn't require coordination between control plane components,
+the changes to each controller are self-contained.
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `CronJobCreationAnnotation`
+  - Components depending on the feature gate: `kube-controller-manager`
+- [ ] Other
+  - Describe the mechanism: N/A.
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No
+  - Will enabling / disabling the feature require downtime or re-provisioning of a node? No
+
+###### Does enabling the feature change any default behavior?
+
+The jobs newly created by cronjob controller will contain a new annotation `CronJobsScheduledAnnotation`.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. If the feature gate is disabled, the CronJob controller will not add the
+scheduled timestamp as an annotation.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The CronJob controller will begin adding the scheduled timestamp as an annotation to jobs created while the feature is enabled, and existing jobs will be unaffected.
+
+###### Are there any tests for feature enablement/disablement?
+
+We plan to add unit tests.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+This change will not impact the rollout or rollback fail. It also will not impact the already running workloads.
+
+###### What specific metrics should inform a rollback?
+
+- Users can monitor CronJobs metrics `job_creation_skew_duration_seconds` and `cronjob_controller_rate_limiter_use`, `cronjob_job_creation_skew`.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+The feature will be tested manually prior to beta launch.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
```

Comment:
Please add

### Monitoring Requirements

[somehow it got removed]

Reply:
This has been removed based on your feedback [here](https://github.com/kubernetes/enhancements/pull/4027#discussion_r1214212802).

Reply:
Commented there - I only meant "N/A" there - not the section title :)